### PR TITLE
Update devDependencies and regenerate package-lock.json

### LINKED ISSUE #401: Update for Stylelint 12.0.0

### PR DESCRIPTION
**Pull request by XhmikosR**
* @babel/cli 7.7.0
* @babel/core 7.7.2
* @babel/preset-env 7.7.1
* babel-preset-jest 24.9.0
* coveralls 3.0.7
* eslint 6.6.0
* husky 3.1.0
* jest 24.9.0
* jest-cli 24.9.0
* lint-staged 9.4.3
* postcss 7.0.21
* prettier 1.19.1
* stylelint 12.0.0

Fixes #401

Closes #396 by superseding it

### CODE REVIEW

**Comment by XhmikosR** [comment]
@kristerkari not super familiar with jest so please confirm these are fine.

BTW you should enable automated security fixes for the repo and optionally dependabot to handle the updates. :)

File: package.json [hunk 1]
```diff
@@ -19,32 +19,32 @@
...
+    "prettier": "^1.19.1",
     "request": "^2.88.0",
     "rimraf": "^3.0.0",
     "semver": "^6.3.0",
-    "stylelint": "^11.0.0"
+    "stylelint": "^12.0.0"
   },
   "peerDependencies": {
-    "stylelint": "^8.0.0 || ^9.0.0 || ^10.0.0 || ^11.0.0"
+    "stylelint": "^10.1.0 || ^11.0.0 || ^12.0.0"
```
Comment (XhmikosR) [inline package.json]: This should have changed before since stylelint requires this.

**Comment by XhmikosR** [comment]
Note that this results in -274 packages and fixes all, minus one moderate, npm vulnerabilities.

The repo should have the automated security fixes enabled for sure. In fact all of your repos should have it enabled @kristerkari :)

File: package.json (hunk 1 above)
Comment (kristerkari) [reply package.json]: `stylelint-scss` should be compatible all the way to version 8, since I don't think that there has been any breaking changes.

Comment (XhmikosR) [reply package.json]: But stylelint itself sets this to `>=10.1.0` so how is this going to work? :)

Comment (kristerkari) [reply package.json]: > how is this going to work?

It's quite simple actually, the peer dep there is separate from stylelint. If you use the newest version of stylelint, then the requirement might be `>=10.1.0`, but if you use an older version stylelint, then the requirement is lower, but here we can support versions 8-12.

Comment (XhmikosR) [reply package.json]: Your call :)

I reverted this and also updated the husky hook to get rid of a new deprecation warning.

Comment (kristerkari) [reply package.json]: Maybe the support for stylelint < 10 could be dropped next time that there is a breaking change to stylelint-scss.

**Comment by XhmikosR** [comment]
@kristerkari can we get this merged? The number of vulnerabilities is pretty insane :)

>found 25408 vulnerabilities (3 moderate, 25402 high, 3 critical)

Hence why I suggested that you enable at least automated security fixes for your repos

**Comment by kristerkari** [comment]
Looks good, only needs rebasing now that I merged other PRs.

**Comment by XhmikosR** [comment]
Done

**Comment by kristerkari** [comment]
Great, thanks!

**Comment by XhmikosR** [comment]
@kristerkari have you enabled automated security fixes for the repo? Also, have you thought about adding lgtm for dependency management?

**Pull request merged**